File: Sorting/Selectionsort/selectionsort.pl
min([X],X) :- !.
min([X|Xs],X) :- min(Xs,R), X =< R, !.
min([X|Xs],R) :- min(Xs,R), X > R.

remove(_,[],[]).
remove(X,[X|Xs],Xs).
remove(X,[Y|Xs],[Y|R]) :- X \= Y, remove(X,Xs,R). 

selectionsort([],[]) :- !.
selectionsort(Xs,[M|R]) :- min(Xs,M), 
                           remove(M,Xs,I), 
						   selectionsort(I,R).

File: Sorting/Selectionsort/SelectionSort.java
public class SelectionSort {
    public static int [] selectionSort(int [] arr) {
        for(int sorted = 0; sorted < arr.length - 1; sorted++) {
            int min = sorted;
            for(int index = sorted + 1; index < arr.length; index++) {
                if(arr[index] < arr[min]) {
                    min = index;
                }
            }
            int temp = arr[sorted];
            arr[sorted] = min;
            arr[min] = temp;
        }
        return arr;
    }
}
